Back/src/main/java/ua/nm/app/controllers/WeatherController.java

```diff
@@ -15,7 +16,12 @@
 public class WeatherController {
     WeatherModel weather = new WeatherModel();
     LocationModel location = new LocationModel();
-    LocationFinder locationFinder = new LocationFinder();
+    LocationFinder locationFinder;
```

Comment:
Is it a good practice to create class object without instant initialization?

Reply:
It's just a declaration of a variable. The initialisation is in Constructor (row 22). This object will be injected by Spring Framework (@Autowired annotation).

Reply:
Yep, I saw initialization just asked because in C++ is not recommended to create a variable without instant initialization in order to prevent it`s usage before initialization (there is a chance that that in this memory adress that is allocated to variable is stored some garbage and compiler could assume that it is actual data that we want to use) so it is recommended to initially assign variable with null if we want to initialize it further in code.

Reply:
Let me now if it`s acceptable in java

Reply:
It's ok in Java. Spring Autowired annotation garantee you that value will be written into the variable on Dependency Injection step of Bean lifesycle (see img). 
If we talk about Java Core, it has default values for all data types like 0 for int or null for class objects, so you can do it anyway.

<img width="909" alt="Bean-Life-Cycle-Process-flow3" src="https://github.com/user-attachments/assets/95e3911c-529c-4594-b3db-ad1bef60e7c6">